lib/Doctrine/ORM/Proxy/Factory/ProxyFactory.php

```diff
@@ -56,11 +57,6 @@ public function generateProxyClasses(array $classMetadataList) : int;
     /**
      * Gets a reference proxy instance for the entity of the given type and identified by
      * the given identifier.
-     *
-     * @param string $className
-     * @param array  $identifier
-     *
-     * @return \Doctrine\ORM\Proxy\Proxy
      */
-    public function getProxy(string $className, array $identifier) : Proxy;
+    public function getProxy(string $className, array $identifier) : GhostObjectInterface;
```

Comment:
Should probably become `ClassMetadata $class` to avoid repeating metadata lookup operations

Reply:
Not changing this for now - can be done later, eventually

Reply:
Do it now and make it fully finished! Unless you plan to make subsequent PRs on the funcionality.

Reply:
I'll dig in today 👍